DataGrid: add Overview (#6453)
DevExpress/devextreme-documentation: concepts/05 UI Components/DataGrid/03 Overview/10 Key Features.md

- **Data Binding**    
DataGrid can load and update data from different [data source](/Documentation/Guide/Data_Binding/Specify_a_Data_Source/Local_Array/) types. It handles [large data sets](/Documentation/Guide/UI_Components/DataGrid/Enhance_Performance_on_Large_Datasets/) and [remote data](/Documentation/Guide/UI_Components/DataGrid/Enhance_Performance_on_Large_Datasets/#Remote_Operations) efficiently.

- **Various Edit Modes**    
[Edit](/Documentation/Guide/UI_Components/DataGrid/Editing/) DataGrid data in multiple modes, including [form](/Documentation/Guide/UI_Components/DataGrid/Editing/#User_Interaction/Form_Mode) and [popup](/Documentation/Guide/UI_Components/DataGrid/Editing/#User_Interaction/Popup_Mode) modes.

- **Filtering and Sorting**    

> Set up a [filter row](/Documentation/Guide/UI_Components/DataGrid/Filtering_and_Searching/#Filter_Row) and a [header filter](/Documentation/Guide/UI_Components/DataGrid/Filtering_and_Searching/#Header_Filter) to filter data by column values, or use other [approaches to filtering](/Documentation/Guide/UI_Components/DataGrid/Filtering_and_Searching/). [Sort](/Documentation/Guide/UI_Components/DataGrid/Sorting/#User_Interaction) DataGrid by single or multiple columns initially or at runtime.

Reviewer: ```suggestion
Set up a [filter row](/Documentation/Guide/UI_Components/DataGrid/Filtering_and_Searching/#Filter_Row) and a [header filter](/Documentation/Guide/UI_Components/DataGrid/Filtering_and_Searching/#Header_Filter) to filter data by column values, or use other [ways to filter](/Documentation/Guide/UI_Components/DataGrid/Filtering_and_Searching/). You can [sort](/Documentation/Guide/UI_Components/DataGrid/Sorting/#User_Interaction) DataGrid by single or multiple columns initially or at runtime.
```

- **Grouping**    
You can use a column header's context menu or a group panel to [group data](/Documentation/Guide/UI_Components/DataGrid/Grouping/#User_Interaction) in DataGrid. 

- **Rich Customization**    
DataGrid offers extensive [appearance](/Documentation/Guide/UI_Components/DataGrid/Customize_the_Appearance/) and behavior customization.

- **Total and Group Summaries**    

> Apply summaries to aggregate data by columns. Use a [total summary](/Documentation/Guide/UI_Components/DataGrid/Summaries/Total_Summary/) to aggregate all data and a [group summary](/Documentation/Guide/UI_Components/DataGrid/Summaries/Group_Summary/) to aggregate data on group-by-group basis.

Reviewer: ```suggestion
Apply summaries to aggregate data by columns. Use a [total summary](/Documentation/Guide/UI_Components/DataGrid/Summaries/Total_Summary/) to aggregate all data and a [group summary](/Documentation/Guide/UI_Components/DataGrid/Summaries/Group_Summary/) to aggregate data on a group-by-group basis.
```
